unreleased/add-replace-key-all-patterns.md

```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
```

Comment:
Since went with changing the function definition this is a breaking change. Since we aren't doing this breaking change via a deprecation first, this will break the transform processor's integration tests.  As a result, we have to update this PR again to update the transform processor at the same time.  Sorry for the back and forth!

Reply:
Done